### PR TITLE
otelmongo semconv 1.21.0 bump

### PR DESCRIPTION
**Pull request by prestonvasquez**
Bump otelmongo semconv version to 1.21.0

### CODE REVIEW

**Comment by prestonvasquez** [comment]
@pellared @dmathieu PR to bump semconv to 1.24.0

**Comment by MrAlias** [comment]
Please address CI failures.

File: instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/mongo.go [hunk 1]
```diff
@@ -39,9 +39,9 @@ func (m *monitor) Started(ctx context.Context, evt *event.CommandStartedEvent) {
 		semconv.DBSystemMongoDB,
 		semconv.DBOperation(evt.CommandName),
 		semconv.DBName(evt.DatabaseName),
-		semconv.NetPeerName(hostname),
-		semconv.NetPeerPort(port),
-		semconv.NetTransportTCP,
+		semconv.NetPeerName(hostname), //nolint:staticcheck
+		semconv.NetPeerPort(port),     //nolint:staticcheck
+		semconv.NetTransportTCP,       //nolint:staticcheck
```
Comment (MrAlias) [inline instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/mongo.go:44]: These are deprecated semantic conventions. Ignoring a lint check does not seem appropriate here.

Comment (prestonvasquez) [reply instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/mongo.go:44]: We can't change these conventions without creating a breaking change. And multiple semantic conventions for a major version [is not supported](https://github.com/open-telemetry/semantic-conventions/issues/1502). The 1.24.0 bump was but it was requested on #6172. See that PR for more information.

Comment (prestonvasquez) [reply instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/mongo.go:44]: Those conventions may have been introduced in 1.21.0, but updating would create breaking changes. Resolving that is the purpose of #6172, where 1.26.0 removed the deprecated conventions. Please see that ticket, as the issue it attempts to resolve clearly states the migration constraints: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/6171 .

Comment (prestonvasquez) [reply instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/mongo.go:44]: See this branch to see the results: https://github.com/prestonvasquez/opentelemetry-go-contrib/tree/otelmongo-bump-semconv-depr-demo

Comment (MrAlias) [reply instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/mongo.go:44]: `otelmongo` is not a stable API. We should provide a migration path, but there is not a requirement to maintain backwards compatibility here.

+1 to what @damemi suggests: upgrade to 1.21 and work on a migration strategy from there.

Comment (prestonvasquez) [reply instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/mongo.go:44]: @MrAlias otelmongo may not be stable, but what you're asking is still a breaking change. In which case we should just re-open and merge #6172. 

@pellared [specifically asked](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/6172#pullrequestreview-2347227726) that we bump to 1.24.0 in #6172.

Comment (MrAlias) [reply instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/mongo.go:44]: > Do we want to upgrade to 1.21.0 instead?

Yes.

I imagine @pellared had mentioned 1.24 as that is the compatible version for HTTP and it was assumed the same for messaging.

Comment (MrAlias) [reply instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/mongo.go:44]: > otelmongo may not be stable, but fixing these deprecations is still a breaking change.

Understood. Hence my comments on thinking though a migration path, but ultimately complying with semver in that this is not a stable API and no stability guarantee has been made.

Comment (prestonvasquez) [reply instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/mongo.go]: Updated! Thanks for the guidance.

**Review by dmathieu — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Transform nil attribute values to `log.Value` zero value instead of `log.StringValue("<nil>")` in `go.opentelemetry.io/contrib/bridges/otelslog`. (#6246)
 - Fix `NewClientHandler` so that `rpc.client.request.*` metrics measure requests instead of responses and `rpc.client.responses.*` metrics measure responses instead of requests in `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc`. (#6250)
 - Fix issue in `go.opentelemetry.io/contrib/config` causing `otelprom.WithResourceAsConstantLabels` configuration to not be respected. (#6260)
+- Bump otelmongo semconv from 1.17.0 to 1.21.0. (#6272)
```
Comment (pellared) [inline CHANGELOG.md:25]: 1. This should not be in "Fixed". We these kind of changes to `Changed`.
2. We use full import paths.

**Review by MrAlias — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Pull request merged**